hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -7025,19 +7025,23 @@ String prepareBulkLoad(byte[] family, String srcPath, boolean copyFile)
       }
 
       isSuccessful = true;
-      //request compaction
-      familyWithFinalPath.keySet().forEach(family -> {
-        HStore store = getStore(family);
-        try {
-          if (this.rsServices != null && store.needsCompaction()) {
-            this.rsServices.getCompactionRequestor().requestCompaction(this, store,
-              "bulkload hfiles request compaction", Store.PRIORITY_USER + 1,
-              CompactionLifeCycleTracker.DUMMY, null);
+      if (conf.getBoolean(HConstants.COMPACTION_AFTER_BULKLOAD_ENABLE, false)) {
```

Comment:
Shouldn't it be enabled by default, to keep consistent with previous behaviour?

Reply:
Yes,but I think it's better to set false by default, this feature will be enable only when user explicitly set it. Otherwise difficult to find this performance effect

Reply:
I agree with @wchevreuil that we'd better keep the old behavior by default, at least on branch-2. We could file a new issue to change it to false on master branch. WDYT?

Reply:
Yes, I totally agree

Reply:
@wchevreuil I have added in ref-guide section.